src/language-js/print/call-arguments.js

```diff
@@ -95,40 +90,42 @@ function printCallArguments(path, options, print) {
     return allArgsBrokenOut();
   }
 
-  const shouldGroupFirst = shouldGroupFirstArg(args);
-  const shouldGroupLast = shouldGroupLastArg(args, options);
-  if (shouldGroupFirst || shouldGroupLast) {
-    if (
-      shouldGroupFirst
-        ? printedArguments.slice(1).some(willBreak)
-        : printedArguments.slice(0, -1).some(willBreak)
-    ) {
+  if (shouldGroupFirstArg(args)) {
+    const tailArgs = printedArguments.slice(1);
+    if (tailArgs.some(willBreak)) {
       return allArgsBrokenOut();
     }
+    let firstArg;
+    const firstArgSelector = isDynamicImport ? "source" : ["arguments", 0];
```

Comment:
Can we put this logic together with `getCallArguments` and `iterateCallArgumentsPath`?